pipeline/astro_cal.py

```diff
@@ -28,11 +149,472 @@ class AstroCalibrator:
+        df = dl.helpers.utils.convert( qresult, "pandas" )
+
+        # Convert this into a FITS file format that scamp would recognize.
+        # In particular, FITS header keywords have to be all upper case
+        # The _WORLD variables are standard.
+        # The others are not global standards, so we'll have to do things
+        # consistent with what we'll do in scamp.
+        # NOTE : the mag error columns are wrong by a factor of 1.09 = 2.5/ln(10)
+
+        df.rename(
+            columns={
+                "ra": "X_WORLD",
+                "dec": "Y_WORLD",
+                "ra_error": "ERRA_WORLD",
+                "dec_error": "ERRB_WORLD",
+                "phot_g_mean_mag": "MAG_G",
+                "phot_g_mean_flux_over_error": "MAGERR_G",
+                "phot_bp_mean_mag": "MAG_BP",
+                "phot_bp_mean_flux_over_error": "MAGERR_BP",
+                "phot_rp_mean_mag": "MAG_RP",
+                "phot_rp_mean_flux_over_error": "MAGERR_RP",
+                "pm": "PM",
+                "pmra": "PMRA",
+                "pmdec": "PMDEC" },
+            inplace=True
+        )
+        # Make the errors actual magnitude errors.  (1.0857 = 2.5/ln(10))
+        for band in [ 'G', 'BP', 'RP' ]:
+            df[ f'MAGERR_{band}' ] = 1.0857 / df[ f'MAGERR_{band}' ]
+        # Put in two more fields that scamp really wants
+        df[ 'OBSDATE' ] = 2015.5
+        df[ 'FLAGS' ] = 0
+
+        # To avoid saving too many files in one directory, use a healpix subdirectory
+        hpix = healpy.ang2pix( 4, ra, dec, lonlat=True )
+        minmagstr = 'None' if minmag is None else f'{minmag:.1f}'
+        maxmagstr = 'None' if maxmag is None else f'{maxmag:.1f}'
+        relpath = ( pathlib.Path( "GaiaDR3_excerpt" ) / str(hpix) /
+                    f"Gaia_DR3_{ra:.4f}_{dec:.4f}_{minmagstr}_{maxmagstr}.fits" )
+        ofpath = pathlib.Path( FileOnDiskMixin.temp_path ) / relpath
+        dbpath = pathlib.Path( FileOnDiskMixin.local_path ) / relpath
+        ofpath.parent.mkdir( parents=True, exist_ok=True )
+
+        fitstab = astropy.table.Table.from_pandas( df )
+        _logger.debug( f"Writing {len(fitstab)} gaia stars to {ofpath}" )
+        ldac.save_table_as_ldac( fitstab, ofpath, overwrite=True )
+
+        catexp = CatalogExcerpt( format='fitsldac', origin='GaiaDR3', num_items=len(fitstab),
+                                 minmag=minmag, maxmag=maxmag, ra=ra, dec=dec,
+                                 ra_corner_00=ralow, ra_corner_01=ralow, ra_corner_10=rahigh, ra_corner_11=rahigh,
+                                 dec_corner_00=declow, dec_corner_10=declow,
+                                 dec_corner_01=dechigh, dec_corner_11=dechigh )
+        catexp.calculate_coordinates()
+
+        return catexp, str( ofpath ), str( dbpath )
+
+    # ----------------------------------------------------------------------
+
+    def secure_GaiaDR3_excerpt( self, image, maxmags=(22.,), magrange=4.,
```

Comment:
The reason I use 'secure' here is that it might not actually fetch it.  In this case, it first checks to see if it's there already, and if so, it just returns what's there.

Reply:
I thought that was the point of fetch, that it is better than get as it allows for lazy downloading something, but only if it isn't there. It is just semantics.